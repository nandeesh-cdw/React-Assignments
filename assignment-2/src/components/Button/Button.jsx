import React from 'react'
import styles from './Button.module.scss';

function Button({name,onButtonClick }) {
  return (
    <button className={styles.button}onClick={onButtonClick}>{name}</button>
  )
}

export default Button