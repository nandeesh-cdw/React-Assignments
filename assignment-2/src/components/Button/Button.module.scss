@import '../../styles/global.scss';

.button{
    cursor: pointer;
    border: none;
    min-width: 158px;
    padding: 15px 25px;
    font-size: 17px;
    font-weight: $font-weight-600;
    color: #fff;
    background-color: $primary-color;
}