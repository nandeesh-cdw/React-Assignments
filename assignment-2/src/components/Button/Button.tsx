import React from 'react'
import styles from './Button.module.scss';

function Button({name,onButtonClick,disabled=false}) {
  return (
    <button className={`${styles.button} ${disabled ? styles.disabled : ''}`}onClick={onButtonClick} disabled={disabled}>{name}</button>
  )
}

export default Button