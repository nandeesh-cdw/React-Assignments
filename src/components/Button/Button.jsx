import React from 'react'
import styles from './Button.module.scss'
export const Button = ({name , selected}) => {
  return (
    <button className={`${styles.button} ${selected ? styles.selected : ''}`}>{name}</button>
  )
}
