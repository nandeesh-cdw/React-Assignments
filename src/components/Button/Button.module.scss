@import '../../styles/global.scss';

button{
    all: unset;
    cursor: pointer;
}
.purple_button{
    font-family: $secondary_font;
    font-weight: $font-weight-500;
    font-size: 13px;
    color: $white;
    border-radius: 20px;
    padding: 10px 25px;
    background-color: $purple;
}
.cyan_button{  
    font-family: $secondary_font;
    font-weight: $font-weight-500;
    font-size: 13px;
    border-radius: 20px;
    padding: 10px 25px;
    background-color: $cyan_blue;
}
.transparent_Button{
    font-family: $secondary_font;
    font-size: 14px;
    color: $black;
    font-weight: $font-weight-500;
}
.purple_button_big{
    font-family: $secondary_font;
    font-weight: $font-weight-500;
    font-size: 19px;
    color: $white;
    border-radius: 25px;
    padding: 10px 50px;
    background-color: $purple;
}