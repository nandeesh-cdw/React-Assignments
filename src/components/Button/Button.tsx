import classNames from 'classnames'
import { useSelector } from 'react-redux'
import styles from './Button.module.scss'
import { ButtonProps, RootState } from '../../models/models'
function Button(props: ButtonProps) {
  const dark_mode = useSelector((state: RootState) => state.navbar.darkMode)
  const buttonStyles = classNames({
    [styles.purple_button]: props.purpleButton,
    [styles.cyan_button]: props.cyanButton,
    [styles.transparent_Button]: props.transparentButton,
    [styles.light_text]: (props.transparentButton && dark_mode),
    [styles.purple_button_big]: props.purpleButtonBig,
  })
  return (
    <button className={buttonStyles} onClick={(event) => props.onClick(event, props.id)}>
      {props.label}
    </button>
  )
}

export default Button