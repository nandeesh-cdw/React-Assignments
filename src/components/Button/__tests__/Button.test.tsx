/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Button from "../Button";

const mockStore = configureStore([]);

test("renders correctly", () => {
  const store = mockStore({ navbar: { darkMode: false } });
  const { asFragment } = render(
    <Provider store={store}>
      <Button label="Click me" onClick={() => {}} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("calls onClick handler when clicked", () => {
  const onClick = jest.fn();
  const store = mockStore({ navbar: { darkMode: false } });
  const { getByText } = render(
    <Provider store={store}>
      <Button label="Click me" onClick={onClick} />
    </Provider>
  );
  // eslint-disable-next-line testing-library/prefer-screen-queries
  const button = getByText("Click me");
  fireEvent.click(button);
  expect(onClick).toHaveBeenCalled();
});

test("renders purple button correctly", () => {
  const store = mockStore({ navbar: { darkMode: false } });
  const { getByText } = render(
    <Provider store={store}>
      <Button label="Click me" purpleButton onClick={() => {}} />
    </Provider>
  );
  const button = getByText("Click me");
  expect(button).toHaveClass("purple_button");
});
