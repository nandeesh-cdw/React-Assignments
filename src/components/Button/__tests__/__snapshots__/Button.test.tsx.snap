// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<DocumentFragment>
  <button
    class=""
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class=""
  >
    Click me
  </button>
</DocumentFragment>
`;
